input/common/wayland_common.h

```diff
@@ -170,7 +170,7 @@ typedef struct gfx_ctx_wayland_data
    struct libdecor *libdecor_context;
    struct libdecor_frame *libdecor_frame;
 #ifdef HAVE_DYLIB
-   struct dylib_t *libdecor;
+   dylib_t libdecor;
```

Comment:
I'm unsure of whether this is correct or not but it fixes a build failure I had. I searched around to see how `dylib_t` was used elsewhere and copied what other code was doing.

```
In file included from ui/drivers/ui_qt.cpp:96:
In file included from ui/drivers/../../gfx/common/wayland_common.h:22:
ui/drivers/../../gfx/common/../../input/common/wayland_common.h:173:11: error: typedef 'dylib_t' cannot be referenced with a struct specifier
  173 |    struct dylib_t *libdecor;
      |           ^
./libretro-common/include/dynamic/dylib.h:46:15: note: declared here
   46 | typedef void *dylib_t;
      |               ^
```